zilliqa/src/db.rs

```diff
@@ -235,12 +354,21 @@ impl Db {
         Ok(sqlite_tx.commit()?)
     }
 
-    pub fn put_canonical_block_number(&self, number: u64, hash: Hash) -> Result<()> {
-        self.block_store.lock().unwrap().execute("INSERT OR REPLACE INTO main_chain_canonical_blocks (height, block_hash) VALUES (?1, ?2)",
+    pub fn set_canonical_block_number_with_db_tx(
```

Comment:
Idle thought: I considered reducing boilerplate by using a macro to generate all the `_with_db_tx` pairs of functions, but haven't spent the time on it. Might be worth to render the file more readable.